py_test.py

```
@@ -0,0 +1,374 @@
+
+    def button_scrap_raw_manufacture(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id("stock.action_stock_scrap")
+        action['domain'] = [('raw_mrp_id', '=', self.id)]
+        action['context'] = dict(self._context, default_origin=self.name)
+        return action
+
+    def action_pick_components(self):
+        """Create a stock picking from main stock to pre-production location."""
+        if not self.raw_move_line_ids:
+           raise ValidationError('Please add component for pick')
```

Flake8 (7.2.0)

[error] indentation is not a multiple of 4

(E111)